src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,343 @@
+'''
+A module to load annotation files for Sentinel-1 IW SLC SAFE data
+'''
+
+import datetime
+import xml.etree.ElementTree as ET
+import os
+import zipfile
+import glob
+import numpy as np
+
+
+
+class Annotation:
+    '''A parent class for other kinds of .xml files. Contains core attributes and common methods.
+       Accepted format: SAFE .zip file, SAFE directory, or annotation .xml file
+    '''
+    # A parent class for annotation reader for Calibrarion, Noise, and poteltially L1
+    def __init__(self,path_annotation:str,polarization='VH',subswath=1):
+        #Attributes
+        self._ipf_version=None
+        self.path_annotation=path_annotation 
+        self.polarization=polarization
+        self.subswath=subswath #Subswath of the annotation file to load
+        self._xml_loaded=None
+        self.kind=''
+
+        #Data access point for the annotation .xml file
+        self.xml_et=None
+
+        # TODO: Detect the expected polarizations from SAFE path
+
+    def _get_search_phrase(self):
+        str_safe=os.path.basename(self.path_annotation).replace('.zip','')
+        token_safe=str_safe.split('_')
+        if len(token_safe)<3:
+            raise ValueError(f'Invalid path_annotation: {self.path_annotation}')
+        else:
+            phrase_base='{PLATFORM}-{MODE}{SUBSWATH}-{TYPE}-{POL}'.format(PLATFORM=token_safe[0].lower(),
+                    MODE=token_safe[1].lower(),
+                    SUBSWATH=self.subswath,
+                    TYPE=token_safe[2].lower(),
+                    POL=self.polarization.lower())
+
+            if self.kind=='calibration' or self.kind=='noise':
+                phrase_xml_to_search=f'{self.kind}-{phrase_base}'
+            elif self.kind=='':
+                phrase_xml_to_search=f'/{phrase_base}'
+            else:
+                raise ValueError(f'Cannot recognize the annotation type {self.kind}')
+
+            return phrase_xml_to_search
+
+
+    def _load_xml_as_et(self): #kind: ['', 'calibration', 'noise']
```

Comment:
In line with the `dataclass` refactor comment, this would be a `classmethod` and all attributes would be populated here as well.

Reply:
A `@classmethod` method `from_et(cls,et_in=None)` in each subclass will replace `_load_xml_as_et`